### PR TITLE
Pin coverage to fix CI

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

The newly released 6.3.1 is still sporadically failing from https://github.com/nedbat/coveragepy/issues/1310

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [n/a] New features are documented, with examples if plot related.
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [n/a] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by QuLogic** [comment]
For Azure, it looks like the macOS images were updated from 20220124.1 to 20220201.2. I'm not seeing anything specifically about Ghostscript in [the changes](https://github.com/actions/virtual-environments/commit/ddce3e81aeeec8c0cb5dbb3f480f178dc4e67199). Ghostscript seems to be installed via Homebrew and hasn't changed versions there, but possibly some of its dependencies have.

**Comment by QuLogic** [comment]
Based on the output from Azure, it appears that Ghostscript is compressing some streams with LZW in binary. This gets somewhat mangled by outputting to `StringIO` and then trying to encode it to compare against the `BytesIO` result. I'm not sure we really should be supporting `StringIO`, or at least disable compression if so.

**Comment by anntzer** [comment]
I think I agree that saving distiller output to text streams does not make sense (as the postscript standard explicitly allows arbitrary binary sequences (section 3.14 "Binary encoding details"), we just don't output them ourselves.  I guess the most backcompatible fix is the one you suggest (to disable distillers when outputting to text, perhaps logging a warning in that case).

File: lib/matplotlib/tests/test_backend_ps.py [hunk 1]
```diff
@@ -71,7 +71,13 @@ def test_savefig_to_stringio(format, use_log, rcParams, orientation):
...
         assert not s_buf.closed
         assert not b_buf.closed
-        s_val = s_buf.getvalue().encode('ascii')
+        try:
+            s_val = s_buf.getvalue().encode('ascii')
+        except UnicodeEncodeError:
+            print('string:',
+                  s_buf.getvalue().encode('ascii', errors='backslashreplace'))
+            print('bytes:', b_buf.getvalue())
+            raise
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_backend_ps.py:80]: Can we move this to a separate PR? These are two independent issues and we're not yet clear whether this here is the correct solution #22416. So it's better address it separately.

Pinning coverage reduces the number of failed CI jobs and should go in as soon as possible.

**Review by timhoffm — APPROVED** [review]
Approval limited to the coverage part.

File: lib/matplotlib/tests/test_backend_ps.py (hunk 1 above)
Comment (QuLogic) [reply lib/matplotlib/tests/test_backend_ps.py]: This is not a solution, just some debug testing. I've removed it.

**Comment by tacaswell** [comment]
I generally dislike hard upper pins, but we will want to do a hard-pin on the 3.5.x branch anyway and it sounds like we are not the only people with this issue.

**Comment by tacaswell** [comment]
I'm going to merge this over the GS related failures.

**Comment by QuLogic** [comment]
> I generally dislike hard upper pins, but we will want to do a hard-pin on the 3.5.x branch anyway and it sounds like we are not the only people with this issue.

I'm following the upstream issue so I can test unpin when it's fixed,

**Pull request merged**